cpp/src/arrow/array-test.cc

```diff
@@ -1764,4 +1765,169 @@ TEST(TestRechunkArraysConsistently, Plain) {
   }
 }
 
+class ConcatenateTest : public ::testing::Test {
+ protected:
+  ConcatenateTest()
+      : rng_(seed_),
+        sizes_({0, 1, 2, 4, 16, 31, 1234}),
+        null_probabilities_({0.0, 0.1, 0.5, 0.9, 1.0}) {}
+
+  std::vector<int32_t> Offsets(int32_t length, int32_t slice_count) {
+    std::vector<int32_t> offsets(static_cast<std::size_t>(slice_count + 1));
+    std::mt19937_64 gen(seed_);
```

Comment:
The Mersenne-Twister PRNG engine has been identified as a source of test slowness.  See discussion [here](https://github.com/apache/arrow/pull/3721#issuecomment-467916496).

Reply:
Alright, I'll switch to `std::default_random_engine`